### PR TITLE
Michigan heating credit

### LINKED ISSUE #2312: Michigan Home Heating Credit
Michigan Home Heating Credit Information Page
https://www.michigan.gov/taxes/questions/iit/accordion/heating/home-heating-credit-information-1
Tax Form
https://www.michigan.gov/taxes/-/media/Project/Websites/taxes/Forms/2022/2022-IIT-Forms/MI-1040CR7.pdf?rev=c36e54deae424566bb6abd1cbe8af7cb&hash=B39BA9887DD31395E754CC0641EFB512
Instruction
https://www.michigan.gov/taxes/-/media/Project/Websites/taxes/Forms/2022/2022-IIT-Forms/BOOK_MI-1040CR-7.pdf?rev=d765b421bc7343c0a54793499dc550db&hash=0B10BE7DD0B91D7F609230E8F636EBC8

### PR DESCRIPTION
**Pull request by PavelMakarchuk**
Fixes #2312

### CODE REVIEW

File: policyengine_us/variables/gov/states/mi/tax/income/credits/home_heating/mi_home_heating_credit_eligible.py
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/mi/tax/income/credits/home_heating/mi_home_heating_credit_eligible.py:1]: this part was not included: 
image

File: policyengine_us/variables/household/expense/housing/heating_costs.py [hunk 1]
```diff
@@ -0,0 +1,9 @@
+from policyengine_us.model_api import *
+
+
+class heating_costs(Variable):
```
Comment (MaxGhenis) [inline policyengine_us/variables/household/expense/housing/heating_costs.py:4]:
```suggestion
class heating_expense(Variable):
```

File: policyengine_us/variables/household/expense/housing/heating_costs.py [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+    value_type = float
+    entity = TaxUnit
```
Comment (MaxGhenis) [inline policyengine_us/variables/household/expense/housing/heating_costs.py:6]:
```suggestion
    entity = Person
```

File: policyengine_us/variables/household/expense/housing/heating_costs.py
Comment (MaxGhenis) [inline policyengine_us/variables/household/expense/housing/heating_costs.py:1]: also add `heating_expense_tax_unit`

File: policyengine_us/variables/gov/states/mi/tax/income/credits/home_heating/alternate_credit/mi_alternate_home_heating_credit.py
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/mi/tax/income/credits/home_heating/alternate_credit/mi_alternate_home_heating_credit.py:1]: remove _credit from folder

File: policyengine_us/parameters/gov/states/mi/tax/income/credits/home_heating/alternate/heating_costs/rate.yaml [hunk 3]
```diff
@@ -0,0 +1,14 @@
+description: Michigan multiplies the difference between the reduced heating costs and the reduced adjusted household resources by this rate under the alternate home heating credit computation.
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/mi/tax/income/credits/home_heating/alternate/heating_costs/rate.yaml:1]:
```suggestion
description: Michigan provides a credit for this fraction of reduced heating costs less reduced adjusted household resources under the alternate home heating credit.
```

File: policyengine_us/parameters/gov/states/mi/tax/income/credits/home_heating/alternate/household_resources/rate.yaml [hunk 4]
```diff
@@ -0,0 +1,14 @@
+description: Michigan reduces the heating costs by the following rate of household resoruces under the alternate home heating credit computation.
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/mi/tax/income/credits/home_heating/alternate/household_resources/rate.yaml:1]:
```suggestion
description: Michigan subtracts this fraction of resources from heating expenses under the alternate home heating credit.
```

File: policyengine_us/parameters/gov/states/mi/tax/income/credits/home_heating/credit_percentage.yaml [hunk 5]
```diff
@@ -0,0 +1,20 @@
+description: Michigan multiplies the lerger of the standard or alternate home heating credit amount by this rate.
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/mi/tax/income/credits/home_heating/credit_percentage.yaml:1]:
```suggestion
description: Michigan provides a credit for this fraction of the greater of the standard and alternate home heating credit.
```

File: policyengine_us/parameters/gov/states/mi/tax/income/credits/home_heating/standard/base.yaml [hunk 6]
```diff
@@ -0,0 +1,45 @@
+description: Michigan provides the following home heating standard credit base, based on the number of exemptions.
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/mi/tax/income/credits/home_heating/standard/base.yaml:1]:
```suggestion
description: Michigan provides this base amount of standard home heating credit, based on the number of exemptions.
```

File: policyengine_us/parameters/gov/states/mi/tax/income/credits/home_heating/standard/fpg_rate.yaml [hunk 7]
```diff
@@ -0,0 +1,16 @@
+description: Michigan limits the home heating credit to filers with household resources below the following rate of the federal poverty guidelines.
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/mi/tax/income/credits/home_heating/standard/fpg_rate.yaml:1]:
```suggestion
description: Michigan limits the home heating credit to filers with household resources below this fraction of the federal poverty guidelines.
```

File: policyengine_us/parameters/gov/states/mi/tax/income/credits/home_heating/standard/included_heating_cost_rate.yaml [hunk 8]
```diff
@@ -0,0 +1,18 @@
+description: Michigan multiplies the home heating credit amount by this rate if the heating costs are included in the filers rent payments.
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/mi/tax/income/credits/home_heating/standard/included_heating_cost_rate.yaml:1]:
```suggestion
description: Michigan multiplies the standard home heating credit by this fraction if rent includes heating.
```

File: policyengine_us/parameters/gov/states/mi/tax/income/credits/home_heating/standard/included_heating_cost_rate.yaml [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+values:
+  2021-01-01: 0.5
+metadata:
+  period: year
+  unit: /1
+  label: Michigan standard home heating credit rate if heating costs are included in rent
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/mi/tax/income/credits/home_heating/standard/included_heating_cost_rate.yaml:7]:
```suggestion
  label: Michigan standard home heating credit rate if rent includes heating
```

File: policyengine_us/parameters/gov/states/mi/tax/income/credits/home_heating/standard/reduction_rate.yaml [hunk 10]
```diff
@@ -0,0 +1,14 @@
+description: Michigan reduces the heating credit amount by this percentage of household resources.
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/mi/tax/income/credits/home_heating/standard/reduction_rate.yaml:1]:
```suggestion
description: Michigan reduces the standard heating credit amount by this fraction of household resources. 
```

**Review by MaxGhenis — CHANGES_REQUESTED** [review]

File: policyengine_us/parameters/gov/states/mi/tax/income/household_resources.yaml
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/mi/tax/income/household_resources.yaml:1]: move this also into the heating credit folder

File: policyengine_us/variables/gov/states/mi/tax/income/mi_household_resources.py [hunk 11]
```diff
@@ -0,0 +1,15 @@
+from policyengine_us.model_api import *
+
+
+class mi_household_resources(Variable):
```
[the remark this answers is not in the document]
Comment (PavelMakarchuk) [reply policyengine_us/variables/gov/states/mi/tax/income/mi_household_resources.py:4]: the household resources definition is applied across multiple programs and is not just defined for the heating credit

File: policyengine_us/parameters/gov/states/mi/tax/income/household_resources.yaml
Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/states/mi/tax/income/household_resources.yaml:1]: the household resources definition is applied across multiple programs and is not just defined for the heating credit

File: policyengine_us/variables/gov/states/mi/tax/income/mi_household_resources.py (hunk 11 above)
[the remark this answers is not in the document]
Comment (MaxGhenis) [reply policyengine_us/variables/gov/states/mi/tax/income/mi_household_resources.py:4]: what other programs? does the MI heating credit legal code point to that central definition?

the law defines it as IRS gross income minus a couple deductions. the form seems to stick to that except adding SSI. the parameter file also includes tanf and general relief, which I don't see mentioned in either source. would be good to back this up with other instances to make sure it's the same thing.

[the remark this answers is not in the document]
Comment (PavelMakarchuk) [reply policyengine_us/variables/gov/states/mi/tax/income/mi_household_resources.py:4]: I also see things like alimony income in the tax form, which I dont think is included in the irs gross income sources 

Should we just use irs_gross_income + the income sources which are not included but mentioned in the form?

I also dont think that the e.g.  business loss deduction currently affects the gross income

[the remark this answers is not in the document]
Comment (MaxGhenis) [reply policyengine_us/variables/gov/states/mi/tax/income/mi_household_resources.py:4]: does the MI heating credit legal code point to that central definition?

[the remark this answers is not in the document]
Comment (PavelMakarchuk) [reply policyengine_us/variables/gov/states/mi/tax/income/mi_household_resources.py:4]: The heating credit [legal code](https://law.justia.com/codes/michigan/2022/chapter-206/statute-act-281-of-1967/division-281-1967-1/division-281-1967-1-9/section-206-527a/) does not offer its own definition but does not reference any specific law (federal or state) as well

**Review by MaxGhenis — APPROVED** [review]

**Pull request merged**